src/main/java/io/harness/cf/client/connector/HarnessConnector.java

```diff
@@ -0,0 +1,187 @@
+package io.harness.cf.client.connector;
+
+import io.harness.cf.ApiClient;
+import io.harness.cf.ApiException;
+import io.harness.cf.api.ClientApi;
+import io.harness.cf.api.MetricsApi;
+import io.harness.cf.model.*;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwt;
+import io.jsonwebtoken.Jwts;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.Request;
+import okhttp3.Response;
+
+@Slf4j
+public class HarnessConnector implements Connector {
+
+  private final ClientApi api;
+  private final MetricsApi metricsApi;
+  private final String apiKey;
+  private final Runnable onUnauthorized;
+  private final HarnessConfig options;
+
+  private String token;
+  private String environment;
+  private String cluster;
+
+  private EventSource eventSource;
+
+  public HarnessConnector(@NonNull String apiKey, Runnable onUnauthorized) {
+    this(apiKey, HarnessConfig.builder().build(), onUnauthorized);
+  }
+
+  public HarnessConnector(
+      @NonNull String apiKey, @NonNull HarnessConfig options, Runnable onUnauthorized) {
+    this.apiKey = apiKey;
+    this.options = options;
+    this.api = new ClientApi(makeApiClient());
+    this.metricsApi = new MetricsApi(makeMetricsApiClient());
+    this.onUnauthorized = onUnauthorized;
+  }
+
+  protected ApiClient makeApiClient() {
+    ApiClient apiClient = new ApiClient();
+    apiClient.setBasePath(options.getConfigUrl());
+    apiClient.setConnectTimeout(options.getConnectionTimeout());
+    apiClient.setReadTimeout(options.getReadTimeout());
+    apiClient.setWriteTimeout(options.getWriteTimeout());
+    apiClient.setDebugging(log.isDebugEnabled());
+    apiClient.setUserAgent("JavaSDK " + io.harness.cf.Version.VERSION);
+    // if http client response is 403 we need to reauthenticate
+    apiClient
+        .getHttpClient()
+        .newBuilder()
+        .addInterceptor(
+            chain -> {
+              Request request = chain.request();
+              // if you need to do something before request replace this
+              // comment with code
+              Response response = chain.proceed(request);
+              if (response.code() == 403 && onUnauthorized != null) {
+                onUnauthorized.run();
+              }
+              return response;
+            })
+        .addInterceptor(new RetryInterceptor(3, 2000));
+    return apiClient;
+  }
+
+  protected ApiClient makeMetricsApiClient() {
+    final int maxTimeout = 30 * 60 * 1000;
+    ApiClient apiClient = new ApiClient();
+    apiClient.setBasePath(options.getEventUrl());
+    apiClient.setConnectTimeout(maxTimeout);
+    apiClient.setReadTimeout(maxTimeout);
+    apiClient.setWriteTimeout(maxTimeout);
+    apiClient.setDebugging(log.isDebugEnabled());
+    apiClient.getHttpClient().newBuilder().addInterceptor(new RetryInterceptor(3, 2000));
```

Comment:
maybe builder or ctor and make ApiClient immutable

Reply:
this class ApiClient is generated from oapi